Events.md

```diff
@@ -0,0 +1,225 @@
+
+Another information that an indexer cannot deduce, is in what scenario the credit event happened. E.g. We don't know if the credit event happened because an offer failed or if the user simply funded Mangrove.
+
+### Debit
+
+This is emitted when a user's account on Mangrove is debited with some native funds.
+
+It emits the `maker`'s address and the `value` debited.
+
+The `maker` address is indexed so that we can filter on it when doing RPC calls.
+
+These are the scenarios where it can happen:
+
+- Withdraw funds from Mangrove directly
+- When posting an offer. The user gets debited the provision that the offer locks.
+- When updating an offer and it requires more provision that it already has. Meaning the user gets debited the difference.
+
+Same challenges as for Credit
+
+### Kill
+
+Emitted when the Mangrove instance is killed
+
+### NewMgv
+
+Emitted when a new Mangrove is deployed.
+
+### OrderStart
+
+This event is emitted when a market order is started on Mangrove.
+
+It emits the `offerlist key`, the `taker`, the `maxLogPrice`, `fillVolume` and `fillWants`.
+
+The fields `offerlist key` and `taker` are indexed, so that we can filter on them when doing RPC calls.
+
+By emitting this an indexer can keep track of what context the current market order is in. E.g. If a user starts a market order and one of the offers taken also starts a market order, then we can in an indexer have a stack of started market orders and thereby know exactly what offerlist the order is running on and the taker.
+
+By emitting `maxLogPrice`, `fillVolume` and `fillWants`, we can now also know how much of the market order was filled and if it matches the price given. See OrderComplete for more.
+
+### OfferFail(WithPosthookData)
+
+This event is emitted when an offer fails, because of a maker error.
+
+It emits `offerlist key`, `taker`, `maker`, the `offerId`, the offers `wants`, `gives`, `penalty` and the `reason` for failure. `Offer list key`, `taker` and `maker` are all fields that we do not need, in order for an indexer to work, as an indexer will be able the get that info from the former `OrderStart` and `OfferWrite` events. But in order for RPC call to filter on this, we need to emit them. Those 3 fields are indexed for the same reason.
+
+If the posthook of the offer fails. Then we emit `OfferFailWithPosthookData` instead of just `OfferFail`. This event has one extra field, which is the reason for the posthook failure. By emitting the posthook data, an indexer can keep track of the reason posthook fails, this could for example be used for analytics.
+
+This event is emitted doring posthook end, we wait to emit this event to the end, because we need the information of `penalty`, which is only available at the end of the posthook. This means that `OfferFail` events are emitted in reverse order, compared to what order they are taken. This is do to the way we handle posthooks. The same goes for `OfferSuccess`.
```

Comment:
```suggestion
This event is emitted doring posthook end, we wait to emit this event to the end, because we need the information of `penalty`, which is only available at the end of the posthook. This means that `OfferFail` events are emitted in reverse order, compared to what order they are taken. This is due to the way we handle posthooks. The same goes for `OfferSuccess`.
```